draft-ietf-idr-5g-edge-service-metadata-24.xml

```diff
@@ -243,15 +246,18 @@
 
       <section title="Ingress Router Forwarding Behavior">
         <t>When the ingress router receives a packet and does a lookup on the
-        route in the FIB, it gets the destination prefix's whole path. It
-        encapsulates the packet destined towards the optimal egress node.</t>
+	route in the FIB, it determines the destination prefix's entire path
+	including the optimal egress node.  The ingress router encapsulates the
+	packet destined towards the optimal egress node.</t>
+	<!-- XXX JMH - this is the first time that encapsulation is mentioned.  Encapsulated how? -->
```

Comment:
The ingress node encapsulates the original packet within a new outer header. The outer header destination address is set to the selected next-hop address

Reply:
Right, but encapsulation implies there's other things going on .  Is there a tunnel encaps attribute?  What happens for non-labeled traffic?

The motivation for asking about this is the inconsistent route selection is a PROBLEM.  Tunneling can get the draft out of that problem, but it needs to be clear what that means.

Reply:
For the non-labeled traffic, no change is needed, i.e., for routes that have Tunnel Encapsulation Path Attributes attached, the ingress routers apply the encapsulation using the Tunnel Attributes associated; and for routes that don't have Tunnel Encapsulation Path Attributes and don't have Metadata Path Attribute, they are forwarded as before. 

The reason for adding the encapsulation header for routes that have Metadata Path Attribute is that intermediate nodes not supporting the metadata path attribute might forward the packet to unintended destinations. 

How about the following sentence? 

"When the ingress router receives a packet and performs a lookup in the FIB, it determines the destination prefix's complete path, including the optimal egress router. For routes that carry the Metadata Path Attribute but lack the Tunnel Encapsulation Path Attribute [RFC9012], it is recommended that the ingress router encapsulate the original packet using an IP-in-IP header. This encapsulation ensures that intermediate nodes not supporting the Metadata Path Attribute do not forward the packet to unintended destinations. The outer header should set the destination address to the optimal egress router and the source address to the ingress router.

For routes without the Metadata Path Attribute, no changes are required—packets are forwarded according to existing behavior, either applying encapsulation when Tunnel Attributes are present or forwarding without encapsulation when they are not."

Reply:
This was addressed in Linda's edits and was merged.  The WG will want to review the text here.